pkg/redirectpolicy/manager.go

```diff
@@ -512,6 +532,9 @@ func (rpm *Manager) getLocalPodsForPolicy(config *LRPConfig, podStore cache.Stor
 		if err != nil {
 			continue
 		}
+		if !rpm.checkPodReady(pod) {
```

Comment:
I don't think this will work as is. If you return here without storing the pods in the `rpm.policyPods` map, you'll never process future update events for the pod? I'll need to think about this a bit more, and I can push a stacked commit to your PR or if you want to give it a shot, feel free to do it.

Reply:
why wouldn't we process update events for the pod? we should be able to detect a ready status change and call `OnUpdatePod` and insert it there, no?

Reply:
I checked the code again, we iterate over all the current policies to see if a pod is selected. I think this should be fine. But please test the case I mentioned offline, where a pod is deployed before an LRP.